### ISSUE TITLE
More specific recommendations in Raise CreateCondaEnvironmentException: Conda must be version 24.7.1 or later, 

### ISSUE THREAD

**Issue by tito-abbo** [snakemake/snakemake#3163]
**Is your feature request related to a problem? Please describe.**
My institution's HPC is running conda version 23.9.0 and does not plan to upgrade any time soon. Thus, the note: "that you can also install conda into the snakemake environment without modifying your main conda installation" seems to be very useful in my situation but after installing the latest conda version within the environment I was using, I still fail the version check for conda. It seems counter intutive to me that I could run a newer version of conda from an older base version, and when I search for how to do this, nothing relevant comes up. 

**Describe the solution you'd like**
If the "note that you can also install conda into the snakemake environment without modifying your main conda installation" does not mean what I think it means, then I think it should be reworded to make the recommendation clearer. If it does indeed mean that you can run a newer version of conda within an environment that has an older version of conda as the base installation, then I think the note should direct the user to some documentation on how to achieve that.

**Comment by mwort** [comment]
I had the same issue and although conda allows you to install any version of itself into the environment (`conda install/update conda`), snakemake doesn't actually pick up the version from the environment. I have only been able to trace it down to the fact snakemake is checking the version using `subprocess.check_output("conda --version", executable=shell_executable)` and without the shell_executable argument it is picking up the correct version. I've resorted to switching off the version check as nothing else seems to do the trick.

**Comment by TarjaTH** [comment]
> I had the same issue and although conda allows you to install any version of itself into the environment (`conda install/update conda`), snakemake doesn't actually pick up the version from the environment. I have only been able to trace it down to the fact snakemake is checking the version using `subprocess.check_output("conda --version", executable=shell_executable)` and without the shell_executable argument it is picking up the correct version. I've resorted to switching off the version check as nothing else seems to do the trick.

May I ask where you switched this off? In one of the rules? I cannot find this in the program file or in the rules of the pipeline I am trying to run. Thank you.

**Comment by erfanshekarriz** [comment]
> May I ask where you switched this off? In one of the rules? I cannot find this in the program file or in the rules of the pipeline I am trying to run. Thank you.

It should be in the source code for snakemake.conda as follows
```python3
from distutils.version import StrictVersion

        try:
            version = shell.check_output(
                self._get_cmd("conda --version"), stderr=subprocess.STDOUT
            ).decode()
            version = version.split()[1]
            if StrictVersion(version) < StrictVersion("4.2"):
                raise CreateCondaEnvironmentException(
                    "Conda must be version 4.2 or later, found version {}.".format(
                        version
                    )
                )
        except subprocess.CalledProcessError as e:
            raise CreateCondaEnvironmentException(
                "Unable to check conda version:\n" + e.output.decode()
            )

```
The file exists in your snakemake conda directory, you can find it by 
```bash
activate snakemake # activate the snakemake directory
which snakemake # prints the path to your snakemake executable file
cd $(python -c "import site; print(site.getsitepackages()[0])")/snakemake # go to source code of executable
pwd # check which directory you're in 
cd deployment # where conda.py file is
head conda.py
```

If you want to switch it off you can simply comment off the lines above in the `conda.py` file with any text editor like nano or vim. 

WARNING: You can change it at your discretion (be careful to remember that you've done this in case you want to reverse it).

**Comment by mwort** [comment]
Yes, I commented the line the `self.check_version` method is called in.

From: Erfan Shekarriz ***@***.***>
Date: Friday, 1 November 2024 at 04:33
To: snakemake/snakemake ***@***.***>
Cc: Michel Wortmann ***@***.***>, Comment ***@***.***>
Subject: Re: [snakemake/snakemake] More specific recommendations in Raise CreateCondaEnvironmentException: Conda must be version 24.7.1 or later, (Issue #3163)

May I ask where you switched this off? In one of the rules? I cannot find this in the program file or in the rules of the pipeline I am trying to run. Thank you.

It should be in the source code for snakemake.conda as follows

from distutils.version import StrictVersion



        try:

            version = shell.check_output(

                self._get_cmd("conda --version"), stderr=subprocess.STDOUT

            ).decode()

            version = version.split()[1]

            if StrictVersion(version) < StrictVersion("4.2"):

                raise CreateCondaEnvironmentException(

                    "Conda must be version 4.2 or later, found version {}.".format(

                        version

                    )

                )

        except subprocess.CalledProcessError as e:

            raise CreateCondaEnvironmentException(

                "Unable to check conda version:\n" + e.output.decode()

            )

The file exists in your snakemake conda directory, you can find it by

activate snakemake # activate the snakemake directory

which snakemake # prints the path to your snakemake executable file

cd $(python -c "import site; print(site.getsitepackages()[0])")/snakemake # go to source code of executable

pwd # check which directory you're in

cd deployment # where conda.py file is

head conda.py

If you want to switch it off you can simply comment off the lines above in the conda.py file with any text editor like nano or vim.

WARNING: You can change it at your discretion (be careful to remember that you've done this in case you want to reverse it).

**Comment by AlcaArctica** [comment]
I have the same problem =/
I installed the latest conda in the sam environment in which I have snakemake 8.28.0 and I also installed the latest conda into all envs which I have specified in the snakemake rules. However, this does not solve the problem. I think when snakemake submits a job to slurm it starts from a new 'session' (?) where it only has the systemwide outdated conda version 23.3.1 available. Then it throws the error: 

```
Building DAG of jobs...
CreateCondaEnvironmentException:
Conda must be version 24.7.1 or later, found version 23.3.1. Please update conda to the latest version. Note that you can also install conda into the snakemake environment without modifying your main conda installation. 
```

So within my snakemake environment it is like this:

```
snakemake -v
8.28.0

conda -V
conda 23.3.1

which conda
/u/luelze/envs/conda_envs/snakemake8/bin/conda

/u/luelze/envs/conda_envs/snakemake8/bin/conda -V
conda 25.1.1
```

I will try the fix proposed above, but his seems very cumbersome.

**Comment by amackays** [comment]
Hi! Just wanted to add that I have the same problem. I'm running snakemake v9.1.6 in an env containing conda v25.3.1 within a SLURM-based system where the base system-wide conda install is v24.1.2.

`CreateCondaEnvironmentException:
Conda must be version 24.7.1 or later, found version 24.1.2. Please update conda to the latest version. Note that you can also install conda into the snakemake environment without modifying your main conda installation.`

The suggested fix above also worked for me, but I agree with other comments that the error message feels misleading & that editing source code like this is cumbersome/not reproducible/risky for users. 

If anyone else runs into this, note that the one-liner provided in this thread for finding the code to edit didn't work on my system for some reason, maybe because I have a custom dir for my envs - the file was in `/path/to/envs/env_name/lib/python3.12/site-packages/snakemake/deployment`.
